Sources/PipecatClientIOS/PipecatClient.swift

```diff
@@ -0,0 +1,902 @@
+
+    public func appendToContext(
+        message: LLMContextMessage,
+        completion: ((Result<AppendToContextResultData, AsyncExecutionError>) -> Void)?
+    ) {
+        Task {
+            do {
+                let response = try await self.appendToContext(message: message)
+                completion?(.success((response)))
+            } catch {
+                completion?(.failure(AsyncExecutionError(functionName: "appendToContext", underlyingError: error)))
+            }
+        }
+    }
+
+    /// Sends a disconnect signal to the bot while maintaining the transport connection.
+    ///
+    /// This method instructs the bot to gracefully end the current conversation session
+    /// and clean up its internal state, but keeps the underlying transport connection
+    /// (WebRTC, WebSocket, etc.) active. This is different from `disconnect()` which
+    /// closes the entire connection.
+    ///
+    /// - Throws:
+    ///   - `BotNotReadyError` if the bot is not in a ready state to accept the disconnect command
+    ///   - Transport errors if the disconnect message fails to send
+    ///
+    /// - Important: The bot must be in a `.ready` state for this method to succeed.
+    /// - Note: This method sends a fire-and-forget message and does not wait for acknowledgment.
+    ///   The bot state change will be reflected through delegate callbacks.
+    /// - SeeAlso: `disconnect()` for closing the entire transport connection.
+    public func disconnectBot() throws {
+        try self.assertReady()
+        try self.sendMessage(
+            msg: .disconnectBot()
+        )
+    }
+
+    /// Registers a function call handler for a specific function name.
+    ///
+    /// When the bot calls a function with the specified name, the registered callback
+    /// will be invoked instead of the delegate's `onLLMFunctionCall` method.
+    ///
+    /// - Parameters:
+    ///   - functionName: The name of the function to handle.
+    ///   - callback: The callback to invoke when this function is called.
+    public func registerFunctionCallHandler(
+        functionName: String,
+        callback: @escaping FunctionCallCallback
+    ) {
+        functionCallCallbacks[functionName] = callback
+    }
+
+    /// Unregisters a function call handler for a specific function name.
+    ///
+    /// After calling this method, function calls with the specified name will
+    /// be handled by the delegate's `onLLMFunctionCall` method instead.
+    ///
+    /// - Parameter functionName: The name of the function to unregister.
+    public func unregisterFunctionCallHandler(functionName: String) {
+        functionCallCallbacks.removeValue(forKey: functionName)
+    }
+
+    /// Unregisters all function call handlers.
+    ///
+    /// After calling this method, all function calls will be handled by
+    /// the delegate's `onLLMFunctionCall` method.
+    public func unregisterAllFunctionCallHandlers() {
```

Comment:
oh good idea. i should probably add the ability to unregister to the web 😬

Reply:
Yep. 🙂